src/liballoc/collections/btree/map.rs

```diff
@@ -200,12 +206,12 @@ impl<K: Clone, V: Clone> Clone for BTreeMap<K, V> {
             }
         }
 
-        if self.is_empty() {
+        if self.is_empty() || self.root.is_none() {
```

Comment:
Good catch. I think I was trying to avoid a panic in the unwrap below, but that doesn't make much sense -- it is true that an empty tree might have a root but we don't need to care :)